src/auth/appConfig.js

```diff
@@ -0,0 +1,105 @@
+
+
+/**
+ * Configuration data for the current app.
+ *
+ * On browser platforms, creating an instance of this
+ * class without any arguments will use
+ * `window.location.origin` as the app domain.
+ * On non-browser platforms, you need to
+ * specify an app domain as the first argument.
+ * @type {AppConfig}
+ */
+export class AppConfig {
+  /**
+   * Blockstack apps are uniquely identified by their app domain.
+   * @type {string}
+   */
+  appDomain: string
+
+  /**
+   * An array of string representing permissions requested by the app.
+   * @type {[Array<string>}
+   */
+  scopes: Array<string>
+
+
+  /**
+   * Path on app domain to redirect users to after authentication. The
+   * authentication response token will be postpended in a query.
+   * @type {string}
+   */
+  redirectPath: string
+
+  /**
+   * Path relative to app domain of app's manifest file.
+   *
+   * This file needs to have CORS headers set so that it can be fetched
+   * from any origin. Typically this means return the header `Access-Control-Allow-Origin: *`.
+   * @type {string}
+   */
+  manifestPath: string
+
+  /**
+   * The URL of Blockstack core node to use for this app. If this is
+   * `null`, the core node specified by the user or default core node
+   * will be used.
+   * @type {string}
+   */
+  coreNode: string
+
+  /**
+   * The URL of a web-based Blockstack Authenticator to use in the event
+   * the user doesn't have Blockstack installed on their machine. If this
+   * is not specified, the current default in this library will be used.
+   * @type {string}
+   */
+  authenticatorURL: ?string
+
+  /**
+   * @param {Array<string>} scopes - permissions this app is requesting
+   * @param {string} appDomain - the app domain
+   * @param {string} redirectPath - path on app domain to redirect users to after authentication
+   * @param {string} manifestPath - path relative to app domain of app's manifest file
+   * @param {string} coreNode - override the default or user selected core node
+   * @param {string} authenticatorURL - the web-based fall back authenticator
+   */
+  constructor(scopes: Array<string> = DEFAULT_SCOPE.slice(),
+              appDomain: string = window.location.origin,
+              redirectPath: string = '',
+              manifestPath: string = '/manifest.json',
+              coreNode: ?string = null,
+              authenticatorURL: string = DEFAULT_BLOCKSTACK_HOST) {
+    this.appDomain = appDomain
+    this.scopes = scopes
+    this.redirectPath = redirectPath
+    this.manifestPath = manifestPath
+
+    if (!coreNode) {
+      this.coreNode = DEFAULT_CORE_NODE
+    } else {
+      this.coreNode = coreNode
+    }
+
+    this.authenticatorURL = authenticatorURL
+  }
+
+  /**
+   * The location to which the authenticator should
+   * redirect the user.
+   * @returns {string} - URI
+   */
+  redirectURI() : string {
+    return `${this.appDomain}${this.redirectPath}`
```

Comment:
Would there ever be a case where we might not want to hardcode the `redirectURI` as a combo `appDomain` and `redirectPath`, eg for native apps or electron apps, I might want to share the data bucket for `coinsapp.co` but use `coins://authRequest` as the redirect for the native implementations. cc @kantai @yknl @hstove

Reply:
@aulneau custom schemes should not be abused for deep links. What is the specification of coins:// scheme? If another app handles this scheme the authentication will fail because it does not know the transit key.